teamscale-gradle-plugin/src/main/kotlin/com/teamscale/TestImpacted.kt

```diff
@@ -18,213 +18,224 @@ import javax.inject.Inject
+
+	val agentJvmConfiguration
+		@Input
+		get() = taskExtension.agent.getAllAgents().map { it.getJvmArgs() }
+
+	val serverConfiguration
+		@Input
+		get() = pluginExtension.server
+
+	/**
+	 * The (current) commit at which test details should be uploaded to.
+	 * Furthermore all changes up to including this commit are considered for test impact analysis.
+	 */
+	val endCommit
+		@Internal
+		get() = pluginExtension.commit.getOrResolveCommitDescriptor(project).first
+
+	/**
+	 *  Can be used instead of [endCommit] by using a revision (e.g. git SHA1) instead of a branch and timestamp.
+	 */
+	val endRevision
+		@Internal
+		get() = pluginExtension.commit.getOrResolveCommitDescriptor(project).second
+
+	/** The baseline. Only changes after the baseline are considered for determining the impacted tests. */
+	val baseline
+		@Input
+		@Optional
+		get() = pluginExtension.baseline
+
+	/**
+	 * Can be used instead of [baseline] by using a revision (e.g. git SHA1) instead of a branch and timestamp
+	 */
+	val baselineRevision
+		@Input
+		@Optional
+		get() = pluginExtension.baselineRevision
+
+	/**
+	 * The repository id in your Teamscale project which Teamscale should use to look up the revision, if given.
+	 * Null or empty will lead to a lookup in all repositories in the Teamscale project.
+	 */
+	val repository
+		@Input
+		@Optional
+		get() = pluginExtension.repository
+
+	/**
+	 * The directory to write the jacoco execution data to. Ensures that the directory
+	 * is cleared before executing the task by Gradle.
+	 */
+	val reportOutputDir
+		@OutputDirectory
+		get() = taskExtension.agent.destination
+
+	/** The report task used to setup and cleanup report directories. */
+	@Internal
+	lateinit var reportTask: TestwiseCoverageReportTask
+
+	val testEngineConfiguration: FileCollection
+		@InputFiles
+		@Classpath
+		get() = project.configurations.getByName(TeamscalePlugin.impactedTestEngineConfiguration)
+
+	init {
+		group = "Teamscale"
+		description = "Executes the impacted tests and collects coverage per test case"
+	}
+
+	@TaskAction
+	override fun executeTests() {
+		val testFrameworkOptions = options
+		require(testFrameworkOptions is JUnitPlatformOptions) { "Only JUnit Platform is supported as test framework!" }
+		if (collectTestwiseCoverage.get()) {
+			require(maxParallelForks == 1) { "maxParallelForks is ${maxParallelForks}. Testwise coverage collection is only supported for maxParallelForks=1!" }
+
+			classpath = classpath.plus(testEngineConfiguration)
+
+			// Workaround to not cause the task to fail when no tests are executed, which might happen when no tests are impacted
+			// https://docs.gradle.org/8.8/userguide/upgrading_version_8.html#test_task_fail_on_no_test_executed
+			// https://github.com/gradle/gradle/blob/master/platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java#L542
+			if (runImpacted) {
+				filter.excludeTestsMatching("dummy-test-name")
```

Comment:
I assume this is the workaround part? 😄

Reply:
Yes. I moved the comments one line closer and added a one more sentence.

Reply:
Thanks. Probably would have been fine one line above, too. Was just a little confused when doing the review with the changes in all the different places. But it's still a little better this way :)